File: WebDriverTests/VS/SeleniumStarter/SeleniumStarter/HomePageTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Firefox;
using Shift.Selenium.Helpers;
using Shift.Selenium.Pages;

namespace Shift.Selenium.Scripts
{
    [TestClass]
    public class HomePageTest
    {
        private IWebDriver driver;
        private HomePage homePage;
        private IWait<IWebDriver> wait;

        [TestInitialize]
        public void TestSetUp()
        {
            FirefoxBinary ffBinary = new FirefoxBinary("C:\\Program Files (x86)\\Mozilla Firefox\\firefox.exe");
            FirefoxProfile ffProfile = new FirefoxProfile();
            driver = new FirefoxDriver(ffBinary, ffProfile);
            homePage = (HomePage)new HomePage(driver);
            homePage.initializeWebElement();
        }

        [TestCleanup]
        public void TestTearDown()
        {
            driver.Quit();
        }

        [TestMethod]
        public void ElementsExistenceAndDefaultValuesTest()
        {
            Assert.IsTrue(homePage.IsTextPresent("Seleniumテスト用ファイル"));
            Assert.AreEqual(homePage.GetInputElementsByType("checkbox").Count, 3);
            Assert.IsFalse(homePage.FemaleRadioboxStatus);
            Assert.IsTrue(homePage.MaleRadioboxStatus);
            Assert.AreEqual(homePage.SelectedCountry, "日本");
        }

        [TestMethod]
        public void ControlLogicTest()
        {
            // check alert
            homePage.EnterName("Kevin Wang");
            homePage.FireNameAlert();
            VerifyAlertWithText("Kevin Wang");
        }


        [TestMethod]
        public void testCalculation()
        {
            homePage.Calculate(2.0, 3.1);
            wait = new WebDriverWait(driver, new TimeSpan(0, 0, 10));
            Assert.AreEqual(homePage.CalculationResult, "5.1");

        }

        [TestMethod]
        public void PageTransitionTest()
        {
            SecondPage secondPage = homePage.MoveToNextPage();
            Assert.IsTrue(secondPage.IsTextPresent("Seleniumテスト用ファイル02"));

        }

        private void VerifyAlertWithText(string text)
        {
            try
            {
                wait = new WebDriverWait(driver, new TimeSpan(0, 0, 2));
                wait.Until(Extensions.AlertIsPresent());
                IAlert alert = driver.SwitchTo().Alert();
                alert.Accept();
                Assert.Equals(alert.Text, text);
            }
            catch (Exception e)
            {
                //exception handling
            }
        }
    }
}

File: WebDriverTests/Eclipse/SeleniumStarter/src/test/java/com/shift/selenium/scripts/HomePageTest.java
package com.shift.selenium.scripts;


import com.shift.selenium.pages.HomePage;
import com.shift.selenium.pages.SecondPage;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.Alert;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Wait;
import org.openqa.selenium.support.ui.WebDriverWait;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

/**
 * Created with IntelliJ IDEA.
 * User: lei.wang
 * Date: 13/11/03
 * Time: 12:36
 * To change this template use File | Settings | File Templates.
 */
public class HomePageTest {
    private WebDriver driver;
    private HomePage homePage;
    private Wait<WebDriver> wait;

    @Test
    public void testElementsExistenceAndDefaultValues() {
        assertThat(homePage.isTextPresent("Seleniumテスト用ファイル"), is(true));
        assertThat(homePage.getInputElementsByType("checkbox").size(), is(3));
        assertThat(homePage.getFemaleRadioboxStatus(), is(false));
        assertThat(homePage.getMaleRadioboxStatus(), is(true));
        assertThat(homePage.getSelectedCountry(), is("日本"));
    }

    @Test
    public void testControlLogic() {
        // check alert
        homePage.enterName("Kevin Wang");
        homePage.fireNameAlert();
        verifyAlertWithText("Kevin Wang");
    }

    @Test
    public void testCalculation() {
        homePage.calculate(2.0, 3.1);
        wait = new WebDriverWait(driver, 10);
        assertThat(homePage.getCalculationResult(), is("5.1"));
    }

    @Test
    public void testPageTransition() {
        SecondPage secondPage = homePage.moveToNextPage();
        assertThat(secondPage.isTextPresent("Seleniumテスト用ファイル02"), is(true));
    }

    private void verifyAlertWithText(String text) {
        try {
            wait = new WebDriverWait(driver, 2);
            wait.until(ExpectedConditions.alertIsPresent());
            Alert alert = driver.switchTo().alert();
            alert.accept();
            assertThat(alert.getText(), is(text));
        } catch (Exception e) {
            //exception handling
        }
    }

    @Before
    public void setUp() {
        driver = new FirefoxDriver();
        homePage = (HomePage) new HomePage(driver).get();
        homePage.initializeWebElement();
    }

    @After
    public void tearDown() {
        driver.quit();
    }
}
